### PR TITLE
PROD-2756: upgrade dependencies and compiler version

### PR DESCRIPTION
**Pull request by jossemargt-cto-ai**
There was an opportunity to get up to date with the latest Rust compiler (1.69.x). To achieve this, I had to update some dependencies and fix some minor complaints from the compiler around our codebase.

### CODE REVIEW

**Review by nitishkumar71 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -28,20 +35,14 @@ Rust libraries. Required parameters are provided to the initial
...
-Unlike the other SDKs, the daemon version in the image is fixed for
-each version of the Bash SDK. If new daemon changes need to be
-propagated to the Bash SDK image, a new version must be created. This
-commit only needs to increment the version number in the Cargo.toml
-and requires no other changes.
+The Bash SDK binaries are automatically built and packaged as `tar.gz` by the
+GitHub Workflows within this repository on every commit. The build logic is
+distributed accross the `scripts/build*` scripts, which make use of
+`docker buildx` to cross compile the binaries, however for local development
+using `cargo build` should suffice.
```
Comment (NickAnderegg) [inline CONTRIBUTING.md:44]:
```suggestion
The Bash SDK binaries are automatically built and packaged as `tar.gz` by the
GitHub Workflows within this repository, on every pushed commit. The build logic is
distributed across the `scripts/build*` scripts, which make use of `docker buildx` to
cross compile the binaries. However, for local development, using `cargo build` should suffice.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -28,20 +35,14 @@ Rust libraries. Required parameters are provided to the initial
...
 
-The image is tagged with tags `<major>-buster-slim`,
-`<major>.<minor>-buster-slim`, and
-`<major>.<minor>.<patch>-buster-slim`, similar to how the daemon
-images are tagged.
+A new release will be cut when the master branch is tagged with the next version
+number, which must match with the version in the root `Cargo.toml`. The
+resulting artifacts will be published within the GitHub **Releases** section.
```
Comment (NickAnderegg) [inline CONTRIBUTING.md:48]:
```suggestion
When the master branch is tagged with a new version (which must match the
version in the root `Cargo.toml` configuration), a new release is built. The
resulting artifacts will be published through the **Releases** of this repo.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
+# Contributing
 
-## Project Structure
+Thanks for taking the time to contribute to SDK Bash!
+
+The SDK Bash is a group of binaries enabling users to interact with the CTO.ai
+API within their workflows. These binaries are written in Rust, so using `cargo`
+for the development cycle should suffice; however, we would like to share some
+details to help you with your contribution.
+
```
Comment (NickAnderegg) [inline CONTRIBUTING.md:9]:
```suggestion
This Bash SDK is a group of binaries enabling users to interact with the CTO.ai
API within their workflows. These binaries are written in Rust, so using `cargo`
for the development cycle should suffice; however, we would like to share some
details to help you with your contribution.

```

**Review by NickAnderegg — CHANGES_REQUESTED** [review]
Some minor edits to the text in CONTRIBUTING.md

**Review by NickAnderegg — APPROVED** [review]
lgtm

**Pull request merged**